posydon/unit_tests/popsyn/test_star_formation_history.py

```diff
@@ -0,0 +1,1059 @@
+        with pytest.raises(ValueError) as excinfo:
+            madau.std_log_metallicity_dist()
+        assert "Invalid sigma value" in str(excinfo.value)
+    
+    def test_csfrd(self):
+        """Test the CSFRD method"""
+        model_dict = {"sigma": 0.5}
+        madau = self.ConcreteMadau(model_dict)
+        
+        # Test with single value
+        z = 0.0
+        result = madau.CSFRD(z)
+        expected = 0.01 * 1**2.6 / (1 + (1/3.2)**6.2)  # a * (1+z)^b / (1 + ((1+z)/c)^d) with z=0
+        np.testing.assert_allclose(result, expected)
+        
+        # Test with array of values
+        z_array = np.array([0.0, 1.0, 2.0])
+        result = madau.CSFRD(z_array)
+        expected = np.array([
+            0.01 * 1**2.6 / (1 + (1/3.2)**6.2),  # z=0 
+            0.01 * 2**2.6 / (1 + (2/3.2)**6.2),  # z=1
+            0.01 * 3**2.6 / (1 + (3/3.2)**6.2)   # z=2
+        ])
+        np.testing.assert_allclose(result, expected)
+    
+    def test_mean_metallicity(self):
+        """Test the mean_metallicity method"""
+        from posydon.utils.constants import Zsun
+        
+        model_dict = {"sigma": 0.5, "Z_max": 0.03}
+        madau = self.ConcreteMadau(model_dict)
+        
+        # Test with single value
+        z = 0.0
+        result = madau.mean_metallicity(z)
+        expected = 10**(0.153) * Zsun
+        np.testing.assert_allclose(result, expected)
+        
+        # Test with array of values
+        z_array = np.array([0.0, 1.0, 2.0])
+        result = madau.mean_metallicity(z_array)
+        expected = 10**(0.153 - 0.074 * z_array**1.34) * Zsun
+        np.testing.assert_allclose(result, expected)
+    
+    def test_fsfr(self):
+        """Test the fSFR method"""
+        model_dict = {"sigma": 0.5, "Z_max": 1}
+        madau = self.ConcreteMadau(model_dict)
+        
+        # Test with redshift array and metallicity bins
+        z = np.array([0.0, 1.0])
+        met_bins = np.array([0.001, 0.01, 0.02, 0.03])
+        
+        result = madau.fSFR(z, met_bins)
+        
+        # Shape check - should be (len(z), len(met_bins)-1)
+        assert result.shape == (2, 3)
+        
+        # THIS IS A VALIDATION TEST
+        from scipy.stats import norm
+        mean0, mean1 = (np.log10(madau.mean_metallicity(z)) 
+                  - model_dict['sigma']**2 * np.log(10) / 2)
+        print(mean0, mean1)
+        
+        expected1 = np.array(
+            # integral from 0.001 to 0.01; Z_min = lowest bin edge
+            [norm.cdf(np.log10(0.01), mean0, model_dict['sigma'])
+             - norm.cdf(np.log10(0.001), mean0, model_dict['sigma']),
+            # integral from 0.01 to 0.02
+             (norm.cdf(np.log10(0.02), mean0, model_dict['sigma'])
+              - norm.cdf(np.log10(0.01), mean0, model_dict['sigma'])),
+            # integral from 0.02 to 1
+             (norm.cdf(np.log10(1), mean0, model_dict['sigma'])
+              - norm.cdf(np.log10(0.02), mean0, model_dict['sigma']))],)
```

Comment:
```suggestion
            [(norm.cdf(np.log10(0.01), mean0, model_dict['sigma'])
              - norm.cdf(np.log10(0.001), mean0, model_dict['sigma'])),
            # integral from 0.01 to 0.02
             (norm.cdf(np.log10(0.02), mean0, model_dict['sigma'])
              - norm.cdf(np.log10(0.01), mean0, model_dict['sigma'])),
            # integral from 0.02 to 1
             (norm.cdf(np.log10(1), mean0, model_dict['sigma'])
              - norm.cdf(np.log10(0.02), mean0, model_dict['sigma']))],)
```
Just for consistency.